### PR TITLE
refactor(FX-3288): Refactor useArtworkFilters and use it on artwork pages

### PR DESCRIPTION
**Pull request by anastasiapyzhik**
The type of this PR is: **Refactor**

This PR resolves [FX-3288]

### Description

In the current implementation, each artworks page (except partner page) duplicates logic of remembering aggregations on first request and refetching data

### Expected format of hook usage:

Instead of

```
useEffect(() => {
  setAggregationsAction(artworks?.aggregations)
}, [])

useEffect(() => {
  if (relay !== undefined && applyFilters) {
    relay.refetchConnection(
      10,
      (error) => {
        if (error) {
            throw new Error("ArtistSeries/ArtistSeriesArtworks filter error: " + error.message)
        }
      },
      { input: prepareFilterArtworksParamsForInput(filterParams) }
    )
  }
}, [appliedFilters])
```

in each artworks page component we can (in most cases) use

```
useArtworkFilters({
  relay,
  aggregations: artworks?.aggregations,
  componentPath: "ArtistSeries/ArtistSeriesArtworks",
  pageSize: 10,
})
```

Refactored hook also takes such optional parameters as custom `refetchVariables`, `onApply` and `onRefetch` callbacks

### PR Checklist (tick all before merging)

- [x] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [x] I have tested my changes on **iOS** and **Android**.
- [x] I have added tests for my changes, or my changes don't require testing, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have added a feature flag, or my changes don't require a feature flag. ([How do I add one?](https://github.com/artsy/eigen/blob/master/docs/developing_a_feature.md))
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/master/docs/adding_state_migrations.md))
- [x] I have added a changelog entry below or my changes do not require one.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

- refactored artwork pages - anastasiapyzhik

</details>

[FX-3288]: https://artsyproduct.atlassian.net/browse/FX-3288

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  ❌ **An error occurred while validating your changelog, please make sure you provided a valid changelog.**
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 327b76ccc32aed993f2c4078a7b21920e60c30b7
</p>

File: src/lib/Components/ArtworkFilter/useArtworkFilters.ts [hunk 1]
```diff
@@ -23,27 +41,29 @@ export const useArtworkFilters = ({
...
+  }, [])
 
   useEffect(() => {
     if (relay !== undefined && applyFilters) {
       relay.refetchConnection(
-        30,
+        pageSize,
         (error) => {
+          if (onRefetch) {
+            onRefetch()
```
Comment (dimatretyak) [inline src/lib/Components/ArtworkFilter/useArtworkFilters.ts:53]: **suggestion**: I think it will be useful to pass error to `onRefetch`

**Review by dimatretyak — APPROVED** [review]
LGTM 👍 Thanks for refactor!

**Review by gkartalis — APPROVED** [review]
Amazing work! 🚀 Thank you so much for this refactor!

**Review by mdole — APPROVED** [review]
Great work @anastasiapyzhik! So much less copy/pasted code!

**Pull request merged**